File: backend/src/main/java/ropold/backend/model/RevealModel.java
package ropold.backend.model;

import java.util.List;

public record RevealModel(
        String id,
        String name,
        List<String> solutionWords,
        List<String> closeSolutionWords,
        Category category,
        String description,
        boolean isActive,
        String appUserGithubId,
        String appUserUsername,
        String appUserAvatarUrl,
        String appUserGithubUrl,
        String imageUrl
) {
}

File: frontend/src/components/model/RevealModel.ts
import {Category} from "./Category.ts";

export type RevealModel = {
    id: string;
    name: string;
    solutionWords: string[];
    closeSolutionWords: string[];
    category: Category;
    description: string;
    isActive: boolean;
    githubId: string;
    imageUrl: string;
}

export const DefaultReveal: RevealModel = {
    id: "",
    name: "Loading....",
    solutionWords: ["true"],
    closeSolutionWords: ["close"],
    category: "ANIMAL",
    description: "",
    isActive: true,
    githubId: "",
    imageUrl: "",
};


